wire-tests/src/test/proto/kotlin/deprecated_enum.proto

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package squareup.protos.kotlin;
+
+option java_package = "com.squareup.wire.protos.kotlin";
+
+enum DeprecatedEnum {
+  DISABLED = 1 [deprecated = true];
+  ENABLED = 2 [deprecated = true];
+  ON = 3;
```

Comment:
Maybe we can add one `[deprecated = false]` or whatever to test the `"true"` check as well?

Reply:
I don't think that `[deprecated = false]` makes sense in practice, but I can add it for the sake of testing.

Reply:
Or then, we could just check that the `deprecated` option is there without checking its value

Reply:
That would be a test for the schema, not for the generator, and I'm pretty sure that test exists. Here we only care about the annotation being generated if the value of `deprecated` is `true`.